### DISCUSSION TITLE
enum cases aren't preserved

### DISCUSSION THREAD

**Discussion by joeprivettsmart** [dotansimha/graphql-code-generator#8856] [Q&A] [answered]
I have an enum in my code that looks like this:

```
export enum Decision {
  already_booked = 'already_booked',
  book_now = 'book_now',
  opt_out = 'opt_out',
}
```

after running the type graphql generator, it ends up like this:

```
export enum Decision {
  AlreadyBooked = 'already_booked',
  BookNow = 'book_now',
  OptOut = 'opt_out',
}
```

Why doesn't it preserve the original case?

**Comment by decaf-dev** [comment, marked as answer]
I just found an answer in the documentation. You need to supply a `namingConvention` key with the value `keep` to your configuration file.

```javascript
import { CodegenConfig } from '@graphql-codegen/cli'
 
const config: CodegenConfig = {
  // ...
  config: { namingConvention: 'keep' }
  // ...
}
export default config
```

Link: https://the-guild.dev/graphql/codegen/docs/config-reference/naming-convention
